cram_3d_world/cram_bullet_reasoning_belief_state/src/process-modules.lisp

```diff
@@ -0,0 +1,89 @@
+;;; POSSIBILITY OF SUCH DAMAGE.
+
+(in-package :cram-bullet-reasoning-belief-state)
+
+(cpm:def-process-module world-state-detecting (motion-designator)
+  ;;(roslisp:ros-info (world-state-sensing) "Tries to sense the new position of the object")
+  (destructuring-bind (command object) (desig:reference motion-designator)
+    (ecase command
+      (cram-common-designators::sense-pose
+       (sense-new-object-pose object)))))
+
+(def-fact-group pr2-matching-pms (cpm:matching-process-module)
+  (<- (cpm:matching-process-module ?motion-designator world-state-detecting)
+    (desig:desig-prop ?motion-designator (:type :world-state-detecting))))
+
+(defun sense-new-object-pose (?old-object)
+  (let (map->base
+        map->obj
+        trans
+        ?newPose
+        ?newTransform
+        new-object)
+
+    (setf map->base (cl-tf:pose->transform (cram-tf:robot-current-pose)))
+    
+    (setf map->obj (second
+                    (find :transform-in-map (desig:desig-prop-value ?old-object :pose)
+                          :test #'equal :key #'first)))
+  
+    (setf trans (cl-transforms:transform* (cl-transforms:transform-inv map->base) map->obj))
+    
+    (setf ?newPose (cl-transforms-stamped:make-pose-stamped
+                    "base_footprint"
+                    0.0
+                    (cl-transforms:translation trans)
+                    (cl-transforms:rotation trans)))
+        
+    (setf ?newTransform (cl-transforms-stamped:make-transform-stamped
+                         "base_footprint"
+                         (desig:desig-prop-value ?old-object :name)
+                         0.0
+                         (cl-transforms:translation trans)
+                         (cl-transforms:rotation trans)))
+    (print ?newTransform)
+                         ;; TODO calculate new Rotation to the object
+                         ;; (cl-transforms:rotation
+                         ;;  (second (second (desig:desig-prop-value ?old-object :pose))))))
+
+    ;; constructing the new Object designator with the new pose and transformation
+    (setf new-object (desig:copy-designator ?old-object
+                                            :new-description
+                                            `((:type ,(desig:desig-prop-value ?old-object :type))
+                                              (:name ,(desig:desig-prop-value ?old-object :name))
+                                              (:pose ((:pose ,?newPose)
+                                                      (:transform ,?newTransform)
+                                                      (:pose-in-map
+                                                       ,(second (third (desig:desig-prop-value ?old-object :pose))))
+                                                      (:transform-in-map
+                                                       ,(second (fourth (desig:desig-prop-value ?old-object :pose)))))))))
```

Comment:
I think we need a utility function that creates the POSE entries for designators. This should go in ``man-int``.
You can skip this comment for now, we can talk about this in person.